### PR TITLE
Add processing to set the event location to the nces school address

### PR DESCRIPTION
**Pull request by drewsamnick**
This will get the location data in the processed_data column for all new form submissions going forward. I still need to figure out the best way to trigger reprocessing all of the existing rows.

### CODE REVIEW

**Comment by drewsamnick** [comment]
[Update HOC map to include matched NCES schools](https://trello.com/c/lW3rUeq4/30-update-hoc-map-to-include-matched-nces-schools)

File: pegasus/forms/hoc_signup_2017.rb [hunk 1]
```diff
@@ -30,6 +30,21 @@ def self.normalize(data)
     result
   end
 
+  def self.process(data)
+    # If there is an nces school id then we need to load that school's address and use it as the event location.
+    # A school id of "-1" indicates that the school was not found in the school dropdown so we treat that the
+    # same as no school id.
+    nces_school_id = data['nces_school_s']
+    if nces_school_id && (nces_school_id != "-1")
```
Comment (breville) [inline pegasus/forms/hoc_signup_2017.rb]: nit: parens necessary?  haven't seen this pattern in our ruby before..

**Comment by drewsamnick** [comment]
During testing I realized that we have not been populating the "organization_name" column of the fusion table with the school_name. This was true even before getting the nces id. I've now added in the code to handle both the manually entered school name and the name of an nces selected school. 

For already processed rows with a school_name_s field this will fix them automatically. For NCES schools this will only handle new rows, just like with the location data.

File: pegasus/forms/hoc_signup_2017.rb (hunk 1 above)
Comment (drewsamnick) [reply pegasus/forms/hoc_signup_2017.rb]: I removed them. My Java brain really wants to put parens around the whole if condition.

File: pegasus/forms/hoc_signup_2017.rb [hunk 2]
```diff
@@ -30,6 +30,23 @@ def self.normalize(data)
...
+  def self.process(data)
+    # If there is an nces school id then we need to load that school's address and use it as the event location.
+    # A school id of "-1" indicates that the school was not found in the school dropdown so we treat that the
+    # same as no school id.
+    nces_school_id = data['nces_school_s']
+    nces_school_name = ''
+    if nces_school_id && nces_school_id != "-1"
+      # id is the primary key of the schools table so we shouldn't get back more than one row
+      DASHBOARD_DB[:schools].where(id: nces_school_id).each do |school_data|
+        school_address = [school_data[:address_line1], school_data[:address_line2], school_data[:address_line3], school_data[:city], school_data[:state], school_data[:zip]].join(' ')
```
Comment (jopolsky) [inline pegasus/forms/hoc_signup_2017.rb]: Would this easier for reading the line?
```
school_address = [:address_line1, :address_line2, :address_line3, :city, :state, :zip].map {|k| school_data[k]}.join(' ')
```
or you can do it multi-line:
```
school_address = [...].map do |key|
  school_data[key]
end.join(' ')
```

File: bin/cron/update_hoc_map [hunk 3]
```diff
@@ -37,7 +37,14 @@ begin
...
       next if form[:review] != "approved" && processed_data['location_city_s'].nil?
 
-      row = [form[:id], data['organization_name_s'], processed_data['location_city_s'], processed_data['location_p'], form[:review], data['special_event_details_s']]
+      if !processed_data['nces_school_name_s'].to_s.strip.empty?
+        organization_name = processed_data['nces_school_name_s']
+      elsif !data['school_name_s'].to_s.strip.empty?
+        organization_name = data['school_name_s']
+      else
+        organization_name = data['organization_name_s']
+      end
```
Comment (jopolsky) [inline bin/cron/update_hoc_map:46]: I've been told to be `!` -averse in ruby, could do something like:
```
organization_name ||= processed_data['nces_school_name_s'].to_s.strip.presence
organization_name ||= data['school_name_s'].to_s.strip.presence
organization_name ||= data['organization_name_s'].to_s.strip.presence
```

Comment (drewsamnick) [reply bin/cron/update_hoc_map:46]: This isn't Rails so I can't use presence. I haven't found a pure Ruby way to evaluate not_empty without ````!````

File: pegasus/forms/hoc_signup_2017.rb (hunk 2 above)
Comment (drewsamnick) [reply pegasus/forms/hoc_signup_2017.rb]: I reworked this with a variation of your suggestion that makes it easier for a reader to parse.

**Pull request merged**